### ISSUE TITLE
Feature request: support typecheck in rule: `valid-expect`

### ISSUE THREAD

**Issue by levchak0910** [vitest-dev/eslint-plugin-vitest#524]
Hi, thanks for such a great plugin, it helps a lot )

Based on [tests](https://github.com/vitest-dev/eslint-plugin-vitest/blob/main/tests/valid-expect.test.ts) seems like the rule `valid-expect` doesn't (or not entirely) supports [typecheck](https://vitest.dev/guide/testing-types)

```ts
import { describe, expectTypeOf, test } from "vitest"

describe("valid-expect", () => {
  test("cases", async () => {
    const fn = (arg: string): number => Number(arg)

    expectTypeOf(fn).toBeFunction() // OK
    expectTypeOf(fn).parameter(0).toMatchTypeOf<string>() // OK
    expectTypeOf(fn).returns.toMatchTypeOf<number>() // Fail: Expect has unknown modifier eslint test/valid-expect
  })
})
```

Do you have any plans to fix or add support for it?
